test/Rings/mpoly-test.jl

```diff
@@ -69,5 +69,27 @@ end
   I = ideal([2*x+3*y+4*z-5,3*x+4*y+5*z-2])
   @test groebner_basis(I,:degrevlex) == [y+2*z-11, 3*x+4*y+5*z-2]
   @test groebner_basis(I,:degrevlex, complete_reduction = true) == [y+2*z-11, x-z+14]
-  
+
+end
+
+@testset "normal_form" begin
+  R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"])
+  I = ideal(R, [x, y])
+  J = ideal(R, [x^2, y*z, z^2])
+  f = x*y^2+x^2+y*z+y^2 + z^2 +z^3
+  @test normal_form(f, I) == z^2 + z^3
+  @test normal_form(J, I) == ideal(R, [z^2])
+end
+
+@testset "Primary decomposition" begin
+  R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"])
+  I = ideal(R, [x, y*z^2])
+  J = ideal(R, [x, y^2])
+  L = PrimaryDecomposition(I)
+  D = Dict(ideal(R, [x, y]) => ideal(R, [x, y]), ideal(R, [x, z^2]) => ideal(R, [x, z]))
+  @test L == D
```

Comment:
What do you mean by robust ? It is using the hash function of Singular for ideals, so I guess it depends on this one...

Reply:
So, Singular.jl makes no attempt to hide the fact that ideals are just (ordered) lists of polynomials (argh!). If the primary decomposition algorithm returns the "different" ideal `ideal(R, [y,x])`, which I see no reason why it can't, then the test is broken. This is what I am talking about:
```
julia> using Singular

julia> R,(x,y)=PolynomialRing(QQ,["x","y"]);

julia> d=Dict{sideal{elem_type(R)}, Int}();

julia> d[Ideal(R,[x,y])]=1;

julia> d[Ideal(R,[y,x])]=2;

julia> d
Dict{sideal{spoly{n_Q}}, Int64} with 2 entries:
  Singular Ideal over Singular Polynomial Ring (QQ),(x,y),(dp(2),C) with generators (x, y) => 1
  Singular Ideal over Singular Polynomial Ring (QQ),(x,y),(dp(2),C) with generators (y, x) => 2
```

Reply:
I think a better test would be to simply test the mathematical defintion of primary decomposition: for each `i` in the dictionary `d`, test that `i` is primary, test `isprime(d[i])`, and test `Singular.equal(d[i], radical(i))`. Finally, test the intersection of all `i` in `d` is `Singular.equal` to `I`.

Reply:
Thank you for the comment and the suggestion ! I will change the test as you suggest.